### ISSUE TITLE
Migration Error: tries to DROP CONSTRAINTS that does not exist.

### ISSUE THREAD

**Issue by Pachat** [directus/directus#22120]
### Describe the Bug

As  #10899 is marked as closed, I have to open this because the problem still occurs and the solutions provided do not work.

The error is
>alter table "directus_fields" drop constraint "directus_fields_collection_foreign" - constraint "directus_fields_collection_foreign" of relation "directus_fields" does not exist

Trying to add it
>ALTER TABLE directus_fields ADD CONSTRAINT directus_fields_collection_foreign FOREIGN KEY (collection) REFERENCES directus_collections (collection);

gives
>ERROR:  Key (collection)=(directus_users) is not present in table "directus_collections".insert or update on table "directus_fields" violates foreign key constraint "directus_fields_collection_foreign" 
ERROR:  insert or update on table "directus_fields" violates foreign key constraint "directus_fields_collection_foreign"
SQL state: 23503
Detail: Key (collection)=(directus_users) is not present in table "directus_collections"

- Suggestion: would be easier if migration would apply

>ALTER TABLE table_name DROP CONSTRAINT IF EXISTS constraint_name;

instead of  only
>ALTER TABLE table_name DROP CONSTRAINT constraint_name;

Is there any patch I could apply to get it?

### To Reproduce

Run `npx directus database migrate:latest ` from 10.2.1 to 10.10.5 on a PostgreSQL DB

### Directus Version

v10.10.5

### Hosting Strategy

Self-Hosted (Custom)

**Comment by Pachat** [comment]
Still present on trying to migrate v10.10.5 to v10.10.7
```
INFO: Skipping CLI extensions initialization due to outstanding migrations.
INFO: Running migrations...
INFO: Applying Remove Collection Foreign Keys...
ERROR: alter table "directus_fields" drop constraint "directus_fields_collection_foreign" - constraint "directus_fields_collection_foreign" of relation "directus_fields" does not exist
    err: {
      "type": "DatabaseError",
      "message": "alter table \"directus_fields\" drop constraint \"directus_fields_collection_foreign\" - constraint \"directus_fields_collection_foreign\" of relation \"directus_fields\" does not exist",
      "stack":
          error: alter table "directus_fields" drop constraint "directus_fields_collection_foreign" - constraint "directus_fields_collection_foreign" of relation "directus_fields" does not exist
      "length": 162,
      "name": "error",
      "severity": "ERROR",
      "code": "42704",
      "file": "tablecmds.c",
      "line": "12056",
      "routine": "ATExecDropConstraint"
    }

```

**Comment by prodkt** [comment]
This is causing a lot of headaches, and still no response. Updates?

**Comment by Awelani-Triba** [comment]
The way I got it work was dropping all the tables related to directus i.e(`directus_*`), and re running `docker composer up`. Not ideal but it got the job done.

Hope this helps.

**Comment by Pachat** [comment]
@Awelani-Triba Thank you for your indication. 
Do you mean that with your `docker-compose.yml` (see [Persistence](https://docs.directus.io/self-hosted/docker-guide.html#persistence)), the tables related to directus come back, 
or do you have some special lines in `docker-compose.yml` to have them back?

**Comment by Pachat** [comment]
This bug could easily be corrected this way:
In file `/node_modules/knex/lib/dialects/postgres/schema/pg-tablecompiler.js` just replace the 3 occurences of
`drop constraint` by `drop constraint if exists`

As this is still not corrected in v10.12.0, how can i modify the `pg-tablecompiler.js` file in docker?

**Comment by Pachat** [comment]
>in file /node_modules/knex/lib/dialects/postgres/schema/pg-tablecompiler.js just replace the 3 occurences of
drop constraint by drop constraint if exists

still not corrected in 10.12.1 so migrate:latest from 10.10.5 to 10.12.1 doesn't work

**Comment by rijkvanzanten** [comment]
I'm not able to reproduce this based on the given reproduction steps. We've updated thousands of postgres-based projects of all sorts of schemas and sizes on our cloud platform and haven't seen this issue, which leads me to believe there's a specific combination of versions, settings, and server environment that causes this.

@Pachat can you share a database dump of a minimal reproduction where you're faced with this issue?

I'll have to close this for now but will happily reopen when more repro information becomes available
